backends/cortex_m/passes/quantized_add_fusion_pass.py

```diff
@@ -0,0 +1,250 @@
+            raise ValueError(f"Cannot extract scalar value from {type(node_arg)}: {node_arg}")
+
+    @staticmethod
+    def _is_qualified_int8_node(args) -> bool:
+        try:
+            if len(args) < 6:
+                return False
+            qmin = int(args[3])
+            qmax = int(args[4])
+            dtype_str = str(args[5])
+            is_int8_range = qmin >= torch.iinfo(torch.int8).min and qmax <= torch.iinfo(torch.int8).max
+            is_int8_dtype = 'int8' in dtype_str.lower()
+            return is_int8_range and is_int8_dtype
+        except (IndexError, ValueError, TypeError):
+            return False
+
+    def call_operator(
+        self,
+        op: EdgeOpOverload,
+        args: tuple,
+        kwargs: dict,
+        meta,
+    ):
+        # Operator replacement logic
+        if op in self.op_replacements:
+            replacement = self.op_replacements[op]
+            if replacement["qualifier"](args):
+                print(f"✅ Replacing operator {op} with {replacement['new_target']}")
+                return super().call_operator(
+                    replacement["new_target"],
+                    args,
+                    kwargs,
+                    meta,
+                )
+            else:
+                print(f"❌ Qualifier failed for operator {op} with args {args}")
+        return super().call_operator(op, args, kwargs, meta)
+
+    def call(self, graph_module: torch.fx.GraphModule):
+        print("🔧 QuantizedAddFusionPass.call() started")
+
+        # STEP 1: Operator replacement
+        replacement_count = 0
+        for node in list(graph_module.graph.nodes):
+            if node.op == 'call_function':
+                print(f"🔍 Checking node: {node.target}")
+
+                if node.target == exir_ops.edge.quantized_decomposed.quantize_per_tensor.default:
+                    print(f"🔍 Found quantized_decomposed.quantize node with args: {node.args}")
+                    if self._is_qualified_int8_node(node.args):
+                        print(f"✅ Replacing quantize node")
+                        node.target = exir_ops.edge.cortex_m.quantize_per_tensor.default
+                        replacement_count += 1
+
+                elif node.target == exir_ops.edge.quantized_decomposed.dequantize_per_tensor.default:
+                    print(f"🔍 Found quantized_decomposed.dequantize node")
+                    if self._is_qualified_int8_node(node.args):
+                        print(f"✅ Replacing dequantize node")
+                        node.target = exir_ops.edge.cortex_m.dequantize_per_tensor.default
+                        replacement_count += 1
+
+                elif node.target == exir_ops.edge.aten.add.Tensor:
+                    print(f"✅ Replacing add node")
+                    node.target = exir_ops.edge.cortex_m.add.Tensor
+                    replacement_count += 1
+
+        print(f"🔧 Operator replacement: Replaced {replacement_count} nodes")
```

Comment:
can we simplify this by using a `subgreph_rewriter`? I am thinking of this as a common pattern for most of these ops.
see - https://docs.pytorch.org/executorch/stable/compiler-custom-compiler-passes.html#level-2